client/fingerprint/fingerprint.go

```diff
@@ -85,6 +85,21 @@ func NewFingerprint(name string, logger *log.Logger) (Fingerprint, error) {
 // Factory is used to instantiate a new Fingerprint
 type Factory func(*log.Logger) Fingerprint
 
+// HealthCheck is used for doing periodic health checks. On a given time
+// interfal, a health check will be called by the fingerprint manager of the
+// node.
+type HealthCheck interface {
+	// Check is used to update properties of the node on the status of the health
+	// check
+	Check(*cstructs.HealthCheckRequest, *cstructs.HealthCheckResponse) error
+
+	// CheckHealthPeriodic is a mechanism for the health checker to indicate that
+	// it should be run periodically. The return value is a boolean indicating
+	// whether it should be done periodically, and the time interval at which
+	// this check should happen.
+	CheckHealthPeriodic() (bool, time.Duration)
```

Comment:
This should be request response as well.

Reply:
I am not a fan of either name.

Check -> HealthCheck
CheckHealthPeriodic -> GetHealthCheckInterval